src/host/main.cpp

```diff
@@ -102,16 +106,34 @@ int main(int argc, char** argv)
     return app.exit(e);
   }
 
-  std::string config_str = files::slurp_string(config_file_path);
+  std::string config_str = files::slurp_string(
+    config_file_path,
+    true /* return an empty string if the file does not exist */);
   nlohmann::json config_json;
-  try
+  auto config_timeout_end = std::chrono::high_resolution_clock::now() +
+    std::chrono::microseconds(config_timeout);
+  std::string config_parsing_error = "";
+  do
   {
-    config_json = nlohmann::json::parse(config_str);
-  }
-  catch (const std::exception& e)
+    std::string config_str = files::slurp_string(
+      config_file_path,
+      true /* return an empty string if the file does not exist */);
+    LOG_INFO_FMT("READ CONFIG {}", config_str);
+    try
+    {
+      config_json = nlohmann::json::parse(config_str);
+      config_parsing_error = "";
```

Comment:
Should there be a `break` in this path? Why are we still looping if parse succeeded?

Reply:
Yes indeed.